### PR TITLE
Add test for [[Description]] from `Symbol.for`

### PR DESCRIPTION
**Pull request by jugglinmike**
Resolves gh-2967. I reviewed the project's coverage for the rest of this algorithm, and it looks okay to me.

### CODE REVIEW

File: test/built-ins/Symbol/for/description.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+esid: sec-symbol.for
+description: Assigns the string value to the [[Description]] slot
+info: |
+    1. Let stringKey be ? ToString(key).
+    [...]
+    4. Let newSymbol be a new unique Symbol value whose [[Description]] value
+       is stringKey.
+    [...]
+    6. Return newSymbol.
+features: [Symbol]
```
Comment (shvaikalesh) [inline test/built-ins/Symbol/for/description.js:13]:
```suggestion
features: [Symbol, Symbol.prototype.description]
```

**Review by shvaikalesh — APPROVED** [review]

**Review by leobalter — CHANGES_REQUESTED** [review]
@jugglinmike The test looks good and it is just missing the feature tag as pointed out by @shvaikalesh.

A small nit pick, thou. I believe this test file should be at [test/built-ins/Symbol/prototype/description](https://github.com/tc39/test262/tree/main/test/built-ins/Symbol/prototype/description) and that folder might be missing other tests to capture description from ToString coerced values.

The decision on the file location might be subjective and it's ok to not go through that path.

**Comment by jugglinmike** [comment]
Thanks for reviewing this!

> just missing the feature tag as pointed out by @shvaikalesh.

All set

> I believe this test file should be at test/built-ins/Symbol/prototype/description

Yeah, the test necessarily uses two different APIs, making the placement a little ambiguous. I chose the `for/` directory because we're specifically interested in how the `[[Description]]` internal slot is defined, not how the `[[Description]]` internal slot is referenced (the existing tests verify the latter).

> that folder might be missing other tests to capture description from ToString coerced values.

I can help with that. I don't see any string coercion in that algorithm, though--could you say more about what you have in mind?

**Comment by rwaldron** [comment]
@leobalter are you good with @jugglinmike's response?

**Comment by leobalter** [comment]
Yes, I just don't have the bandwidth to expand my thoughts on the ToString coercion, but hopefully I can explore it in a near future.

**Comment by jugglinmike** [comment]
Thanks, @leobalter! Would you mind formally registering your approval using the GitHub.com UI? At some point, this became mandatory for Test262 pull requests.

**Review by leobalter — APPROVED** [review]

**Pull request merged**